modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java

```diff
@@ -423,30 +425,96 @@ public void testMap() throws Exception {
     }
 
     /**
+     * Checks deserialization of elements in the singleton map.
+     *
+     * @throws Exception If failed.
+     */
+    public void testSingletonMap() {
+        IgniteCache<Integer, Map<TestObject, TestObject>> c = jcache(0);
+
+        TestObject obj = new TestObject(123);
+
+        c.put(0, Collections.singletonMap(obj, obj));
+        c.put(1, Collections.singletonMap(null, null));
+
+        checkSingleElement(c.get(0).entrySet(), entry -> {
+            assertEquals(123, entry.getKey().val);
+            assertEquals(123, entry.getValue().val);
+        });
+
+        IgniteCache<Integer, Map<BinaryObject, BinaryObject>> kpc = keepBinaryCache();
+
+        Map<?, ?> cBinary = kpc.get(0);
+
+        assertEquals(Collections.singletonMap(null, null).getClass(), cBinary.getClass());
+
+        checkSingleElement(cBinary.entrySet(), entry -> {
+
+            assertTrue(entry.getKey() instanceof BinaryObject);
+            assertTrue(entry.getValue() instanceof BinaryObject);
+            assertEquals(Integer.valueOf(123), ((BinaryObject)entry.getKey()).field("val"));
+            assertEquals(Integer.valueOf(123), ((BinaryObject)entry.getValue()).field("val"));
+        });
+
+        checkSingleElement(kpc.get(1).entrySet(), entry -> {
+
+            assertNull(entry.getKey());
+            assertNull(entry.getValue());
+        });
+    }
+
+    /**
+     * Checks deserialization of elements in the singleton list and set.
+     *
      * @throws Exception If failed.
```

Comment:
Exception is never thrown. It is shown in IDEA.

Reply:
Fixed